src/protocol/ErrorFactory.ts

```diff
@@ -116,17 +113,25 @@ export class ClientErrorFactory {
     }
 
     createErrorFromClientMessage(clientMessage: ClientMessage): Error {
-        const errorCodec = ErrorCodec.decode(clientMessage);
-        return this.createError(errorCodec.errorCode, errorCodec.className, errorCodec.message, null);
+        const errorHolders = ErrorsCodec.decode(clientMessage);
+        return this.createError(errorHolders, 0);
     }
 
-    createError(errorCode: number, className: string, message: string, cause: Error): Error {
-        const factoryFunc = this.codeToErrorConstructor.get(errorCode);
+    createError(errorHolders: ErrorHolder[], errorHolderIdx: number): Error {
+        if (errorHolderIdx === errorHolders.length) {
+            return null;
+        }
+
+        const errorHolder = errorHolders[errorHolderIdx];
+        const factoryFunc = this.codeToErrorConstructor.get(errorHolder.errorCode);
+        let error: Error;
         if (factoryFunc != null) {
-            return factoryFunc(message, cause);
+            error = factoryFunc(errorHolder.message, this.createError(errorHolders, errorHolderIdx + 1));
         } else {
-            return new UndefinedErrorCodeError(message, className);
+            error = new UndefinedErrorCodeError(errorHolder.message, errorHolder.className);
         }
+        // TODO Set error.stack to errorHolder.stackTraceElements
```

Comment:
Is it a leftover or something prevents resolving this TODO within this PR?

Reply:
I created an issue for it to track. https://github.com/hazelcast/hazelcast-nodejs-client/issues/542